gatsby-node.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Implement Gatsby's Node APIs in this file.
+ *
+ * See: https://www.gatsbyjs.org/docs/node-apis/
+ */
+
+// You can delete this file if you're not using it
+
+const { GraphQLJSONObject } = require(`graphql-type-json`)
+const lunr = require(`lunr`)
+
+//NOTE(Rejon): I copy/pasted sanitize filter from our utils file because importing here is different on builds.
+//             I wanted to decouple client facing code from build related code.
+const sanitizeFilter = (rawBody, fragment, normalize) =>
+  rawBody
+    .substring(
+      rawBody.lastIndexOf(`<${fragment}>`) + (fragment.length + 3),
+      rawBody.lastIndexOf(`</${fragment}>`)
+    )
+    .replace(/[-.#`!*()]["]|[0-9]\./g, "") //Replace MD usual characters with empty
+    .split("\n") //Split by new lines
+    .filter(n => n !== "")
+    .map(n => {
+      let label = n.trim()
+
+      //"Normalizing" means removing tildes, hyphens, spaces, and NFD normalization to make it easier to search unique names.
+      //NOTE(Rejon): This is because LUNR is picky about it's token handling. Rejon will show 0 results unless typed Réjon.
+      //             TLDR: Better search experience > exact term searching.
+      if (normalize) {
+        label = label
+          .replace(/\s/g, "")
+          .normalize("NFD")
+          .replace(/[\u0300-\u036f]|-/g, "")
+      }
+
+      return label
+    })
+
+//This is the method that builds our index with Lunr.
+//It's pretty complex, and LUNR has terrible documentation.
+//So please follow this for the BEST documentation offered: https://css-tricks.com/how-to-add-lunr-search-to-your-gatsby-website/
+const createIndex = async (dirNodes, type, cache) => {
+  //NOTE(Rejon): Our index is data handled through graphql queries,
+  //             so we can Cache it! This speeds up builds if none of our data has changed.
+  const cacheKey = `IndexLunr`
+  const cached = await cache.get(cacheKey)
+  if (cached) {
+    return cached
+  }
+  const documents = []
+  const store = {}
+  // Iterate over all posts
+  for (const node of dirNodes) {
+    const { rawBody, id } = node
+    const doc = {
+      name: rawBody
+        .split("\n") //Split by new lines
+        .find(n => n[0] === "#") //Grab name by first H1 "#" that appears
```

Comment:
Non-blocking suggestion - I know the first line that begins with `#` will likely be the `h1` tag, but a space would help filter out other header tags!
```suggestion
        .find(n => n.startsWith("# ")) //Grab name by first H1 "#" that appears
```

Reply:
I agree! I've commited your changes :)